mne_bids/path.py

```diff
@@ -219,38 +254,69 @@ def update(self, **entities):
         If one creates a bids basename using
         :func:`mne_bids.make_bids_basename`:
 
-        >>> bids_basename = make_bids_basename(subject='test', session='two', task='mytask', suffix='data.csv')
+        >>> bids_basename = make_bids_basename(subject='test', session='two',
+                                               task='mytask', kind='data',
+                                               extension='.csv')
         >>> print(bids_basename)
         sub-test_ses-two_task-mytask_data.csv
         >>> # Then, one can update this `BIDSPath` object in place
-        >>> bids_basename.update(acquisition='test', suffix='ieeg.vhdr', task=None)
-        BIDSPath(sub-test_ses-two_acq-test_ieeg.vhdr)
+        >>> bids_basename.update(acquisition='test', kind='ieeg',
+                                 extension='.vhdr', task=None)
+        BIDSPath(root: None
+        basename: sub-test_ses-two_acq-test_ieeg.vhdr)
         >>> print(bids_basename)
         sub-test_ses-two_acq-test_ieeg.vhdr
-        """  # noqa
+        """
         run = entities.get('run')
         if run is not None and not isinstance(run, str):
             # Ensure that run is a string
             entities['run'] = '{:02}'.format(run)
 
+        # ensure extension starts with a '.'
+        ext = entities.get('extension')
+        if ext is not None:
+            if not ext.startswith('.'):
```

Comment:
okay now you call it `ext` here. I would suggest being consistent. I like `ext` better.

Reply:
See: https://github.com/mne-tools/mne-bids/pull/492#discussion_r469707403